src/main/java/org/powerflows/dmn/engine/evaluator/expression/provider/JuelExpressionEvaluationProvider.java

```diff
@@ -16,27 +16,105 @@
 import org.powerflows.dmn.engine.model.decision.field.Input;
 import org.powerflows.dmn.engine.model.decision.rule.entry.InputEntry;
 import org.powerflows.dmn.engine.model.decision.rule.entry.OutputEntry;
 
+import javax.el.ELContext;
+import javax.el.ELException;
+import javax.el.ExpressionFactory;
+import javax.el.FunctionMapper;
 import java.io.Serializable;
 
+@Slf4j
 class JuelExpressionEvaluationProvider implements ExpressionEvaluationProvider {
 
+    private final ExpressionFactory expressionFactory;
+    private final FunctionMapper rootFunctionMapper;
+
+    JuelExpressionEvaluationProvider(final ExpressionEvaluationConfiguration configuration) {
+        this.expressionFactory = new ExpressionFactoryImpl();
+        rootFunctionMapper = new ContextFunctionMapper(configuration.getMethodBinding());
+    }
+
+    private PowerflowsELContext makeContext(final EvaluationContext evaluationContext) {
+        final PowerflowsELContext context = new PowerflowsELContext(rootFunctionMapper);
+        evaluationContext.getAll().forEach((key, value) ->
+            context.setVariable(key, expressionFactory.createValueExpression(value, value == null ? Object.class : value
+                    .getClass()))
+        );
+
+        return context;
+    }
+
+    private ELContext makeContext(final EvaluationContext evaluationContext, final InputEntry inputEntry) {
+        final PowerflowsELContext context = makeContext(evaluationContext);
+        context.setVariable(inputEntry.getNameAlias(), context.getVariableMapper()
+                .resolveVariable(inputEntry.getName()));
+
+        return context;
+    }
+
     @Override
     public Serializable evaluateInput(final Input input, final EvaluationContext evaluationContext) {
-        throw new UnsupportedOperationException();
+        log.debug("Starting evaluation of input: {} with evaluation context: {}", input, evaluationContext);
+
+        final Serializable result = evaluate((String) input.getExpression()
+                .getValue(), getResultType(input), makeContext(evaluationContext));
+
+        log.debug("Evaluated result: {}", result);
+
+        return result;
     }
 
+    private Class<? extends Serializable> getResultType(final Input input) {
+        if (input.getType() != null) {
+            return input.getType().realType();
+        } else {
+            log.warn("No type provided for input {} defaulting to String", input);
+            return String.class;
+        }
+    }
+
+
     @Override
     public Serializable evaluateInputEntry(final InputEntry inputEntry, final EvaluationContext evaluationContext) {
-        throw new UnsupportedOperationException();
+        log.debug("Starting evaluation of input entry {} with evaluation context: {}", inputEntry, evaluationContext);
+
+        final Serializable result = evaluate((String) inputEntry.getExpression()
+                .getValue(), Serializable.class, makeContext(evaluationContext, inputEntry));
```

Comment:
Why here is Serializable.class but in evaluateInput() is getResultType(input)?

Reply:
as there is no elegant way to extract type here, I assume expression author is sane ;)